backstory/src/test/java/com/google/sps/perspective/PerspectiveAPIClientTest.java

```diff
@@ -145,5 +151,34 @@ public void checkRightScores() {
     }
   }
 
-  // TODO: add a test that checks that the correct AnalyzeCommentRequest is passed to PerspectiveAPI
+  /**
+   * Check that the correct AnalyzeCommentRequest is passed to the Perspective API 
+   * by checking all values of AnalyzeCommentRequest that are set by PerspectiveAPIClient.
+   */
+  @Test
+  public void checkRequest() {
+    MockPerspectiveAPIFactory factory = new MockPerspectiveAPIFactory(new HashMap<AttributeType, Float>()); 
+    PerspectiveAPI mockAPI = factory.newInstance();
+    PerspectiveAPIClient client = new PerspectiveAPIClient(mockAPI);
+
+    PerspectiveValues values = client.analyze(DESIRED_TYPES, DEFAULT_TEXT);
+
+    ArgumentCaptor<AnalyzeCommentRequest> requestCaptor = ArgumentCaptor.forClass(AnalyzeCommentRequest.class);
+    verify(mockAPI).analyze(requestCaptor.capture()); 
+
+    AnalyzeCommentRequest actual = requestCaptor.getValue();
+
+    // check that the entry to score and requested attributes are as expected
+    Assert.assertEquals(DEFAULT_TEXT, actual.getComment().getText()); // check text
+    Assert.assertEquals(ContentType.PLAIN_TEXT, actual.getComment().getType()); // check content type
+
+    Map<AttributeType, RequestedAttribute> requestedAttributes = actual.getRequestedAttributes();
+
+    Assert.assertEquals(DESIRED_TYPES.size(), requestedAttributes.size()); // check number of requested types
+    
+    // check that the types requested match
+    for (AttributeType type: DESIRED_TYPES) {
+      Assert.assertTrue(requestedAttributes.containsKey(type));
+    }
```

Comment:
Can the 6 lines above be replaced with something like:

Assert.assertEquals(new HashSet<AttributeType>(DESIRED_TYPES), requestedAttributes.keySet());

Reply:
I can change it. Thanks for the suggestion!

Reply:
Done